Code/GraphMol/Aromaticity.cpp

```diff
@@ -619,30 +648,107 @@ int countAtomElec(const Atom *at) {
 }
 
 namespace {
-// use minRingSize=0 or maxRingSize=0 to ignore these constraints
-int aromaticityHelper(RWMol &mol, unsigned int minRingSize,
-                      unsigned int maxRingSize, bool includeFused) {
-  // FIX: we will assume for now that if the input molecule came
-  // with aromaticity information it is correct and we will not
-  // touch it. Loop through the atoms and check if any atom has
-  // arom stuff set.  We may want check this more carefully later
-  // and start from scratch if necessary
-  ROMol::AtomIterator ai;
-  for (ai = mol.beginAtoms(); ai != mol.endAtoms(); ai++) {
-    if ((*ai)->getIsAromatic()) {
-      // found aromatic info
-      return -1;
+int mdlAromaticityHelper(RWMol &mol, const VECT_INT_VECT &srings) {
+  int narom = 0;
+  // loop over all the atoms in the rings that can be candidates
+  // for aromaticity
+  // Atoms are candidates if
+  //   - it is part of ring
+  //   - has one or more electron to donate or has empty p-orbitals
+  int natoms = mol.getNumAtoms();
+  boost::dynamic_bitset<> acands(natoms);
```

Comment:
perhaps acandidates, acands is not as obvious to me as it should be (even considering the comment above)